osbs/api.py

```diff
@@ -475,7 +475,6 @@ def _do_create_prod_build(self, git_uri, git_ref,
             koji_kerberos_keytab=self.build_conf.get_koji_kerberos_keytab(),
             koji_kerberos_principal=self.build_conf.get_koji_kerberos_principal(),
             architecture=architecture,
-            platform=platform,
```

Comment:
`create_worker_build()` requires it, yes, and passes it to `_do_create_prod_build()`. It requires it because the build request needs to know what node selector to use. This is handled by passing the node selector for that platform directly to the build request: https://github.com/projectatomic/osbs-client/pull/586/files/4430ab1bafc13b01cd5852b6b93b045fbb3a656b#diff-034b3f8b7a38c1ab83d3dc6cce5f2683R508

We don't also need to pass the platform in because it's not used for anything else.